### PR TITLE
Handle line sections

### PR DESCRIPTION
**Pull request by t-ober**
Builds on #141

### CODE REVIEW

**Comment by t-ober** [comment]
!test

File: src/main/scala/edu/ie3/powerFactory2psdm/model/entity/LineSection.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  def buildLineSectionMap(
+      rawLineSections: List[LineSections]
+  ): Map[String, List[LineSection]] = {
+    val lineSections = rawLineSections.map(build)
+    lineSections.groupBy(lineSection => getLineId(lineSection.id))
+  }
+
+  def getLineId(lineSectionId: String): String = {
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/powerFactory2psdm/model/entity/LineSection.scala:50]:
```suggestion
  private def getLineId(lineSectionId: String): String = {
```

File: src/main/scala/edu/ie3/powerFactory2psdm/model/entity/LineSection.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      )
+    )
+    val length = rawLineSection.dline.getOrElse(
+      throw MissingParameterException(
+        s"There is no defined length for line section $id"
+      )
+    )
+    val typeId = rawLineSection.typeId.getOrElse(
+      throw MissingParameterException(
+        s"There is no defined type line section $id"
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/powerFactory2psdm/model/entity/LineSection.scala:37]:
```suggestion
        s"There is no defined type for line section $id"
```

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
looks mostly good to me, just some small suggestions

**Comment by sebastian-peter** [comment]
Oh and btw, I think it would make sense to note these additions and changes in `CHANGELOG.md`

File: src/main/scala/edu/ie3/powerFactory2psdm/converter/LineConverter.scala [hunk 3]
```diff
@@ -44,34 +45,47 @@ object LineConverter {
...
+          getLineType(lineTypeId, lineTypes).getOrElse(
+            throw ConversionException(
+              s"Could not convert line: $line due to failed line type retrieval."
+            )
+          )
+        case (None, None) =>
+          throw ConversionException(
+            s"Could not convert line: ${line.id} since there is no defined type in the model and there are no line section that specify the type"
+          )
+      }
```
[the remark this answers is not in the document]
Comment (t-ober) [reply src/main/scala/edu/ie3/powerFactory2psdm/converter/LineConverter.scala:66]: I think you can do it both ways. I would rather keep it here as this logic is mostly about determining if this is a line with line sections for which we still have to do the conversion for and its not only about retrieving the corresponding line type from the line type mapping.

[the remark this answers is not in the document]
Comment (t-ober) [reply src/main/scala/edu/ie3/powerFactory2psdm/converter/LineConverter.scala:66]: Tbh having different locations where we convert line types is not optimal but has to do with the way our powerfactory grid data is organized. I will document that a little more and then I think this is fine.

**Review by sebastian-peter — APPROVED** [review]

**Pull request merged**